ner_v2/detectors/numeral/number/standard_number_detector.py

```diff
@@ -0,0 +1,247 @@
+        """
+        self.text = ''
+        self.tagged_text = ''
+        self.processed_text = ''
+        self.date = []
+        self.original_date_text = []
+        self.entity_name = entity_name
+        self.tag = '__' + entity_name + '__'
+
+        self.numbers_word = {}
+        self.language_scale_map = {}
+        self.units_map = {}
+
+        # Method to initialise value in regex
+        self.init_regex_and_parser(data_directory_path)
+
+        self.regex_numeric_patterns = re.compile(r'(([\d,]+[.]?[\d]*)\s?(' + "|".join(self.language_scale_map.keys())
+                                                 + r')?)\s*', re.UNICODE)
+
+        # Variable to define default order in which detector will work
+        self.detector_preferences = [self._detect_number_from_digit,
+                                     self._detect_number_from_numerals
+                                     ]
+
+    def detect_number(self, text):
+        self.text = text
+        self.processed_text = text
+        self.tagged_text = text
+
+        number_list, original_list = None, None
+        for detector in self.detector_preferences:
+            number_list, original_list = detector(number_list, original_list)
+            self._update_processed_text(original_list)
+        return number_list, original_list
+
+    @staticmethod
+    def _get_list_from_pipe_sep_string(numerals_text):
+        """
+        Split numerals
+        Args:
+            numerals_text (str): numerals text
+        Returns:
+            (list) : list containing numeral after split
+        """
+        numerals_list = numerals_text.split("|")
+        return [num.lower().strip() for num in numerals_list if num.strip()]
+
+    def init_regex_and_parser(self, data_directory_path):
+        """
+        Initialise numbers word from from data file
+        Args:
+            data_directory_path (str): path of data folder for given language
+        Returns:
+            None
+        """
+        # create number_words dict having number variants and their corresponding scale and increment value
+        # create language_scale_map dict having scale variants and their corresponding value
+        numeral_df = pd.read_csv(os.path.join(data_directory_path, NUMBER_NUMERAL_CONSTANT_FILE), encoding='utf-8')
+        for index, row in numeral_df.iterrows():
+            name_variants = self._get_list_from_pipe_sep_string(row[NUMBER_DATA_FILE_NAME_VARIANTS])
+            value = row[NUMBER_DATA_FILE_VALUE]
+            if float(value).is_integer():
+                value = int(row[NUMBER_DATA_FILE_VALUE])
+            number_type = row[NUMBER_DATA_FILE_TYPE]
+
+            if number_type == NUMBER_TYPE_UNIT:
+                for numeral in name_variants:
+                    self.numbers_word[numeral] = (1, value)
+            else:
+                for numeral in name_variants:
+                    self.numbers_word[numeral] = (value, 0)
```

Comment:
What does this '0' stand for at zero position in tuple? Please put an inline comment

Reply:
Done